### DISCUSSION TITLE
How can I protect my server?

### DISCUSSION THREAD

**Discussion by macrochel** [itzg/docker-minecraft-server#2637] [Q&A]
How can I protect my server from bots which connect to it. I have white-list, but it doesn't help. Maybe I should change the port or something else?

**Comment by dgradecak** [comment]
hi, I am also interested in this. I have the sever accessible via a public IP and thus anyone with a TLauncher can connect easily. Also via tlaucnher (I was testing) it is also simple to change the uuid in tlauncher.

Running the server for my son and it seems some guys from "mojang loss prevention" connected somehow (probably first he disabled the whitelist and left the server without whitelist overnight), and now once the usernames are known, it seems so easy to connect to the server.

**Reply by FreeFallFail** [reply]
“Mojang Loss Prevention Inc.” isn’t affiliated with Mojang or any official loss prevention effort; it’s actually the name a group of Minecraft griefers use for themselves online, partnering with the 5th Column. 

Players who’ve dealt with them describe MLPI as an organized griefing group that actively scans for unprotected or cracked/offline-mode servers, joins using spoofed names or automated tools, destroys builds (often by lavacasting large areas), leaves signs or messages linking to their Discord, and even lures devastated server owners with fake “server restoration” help that’s really just trolling or a scam despite their claims of anti-piracy or “renovation,” they’re widely treated as harassers who target vulnerable worlds and mock their victims rather than help them.

There's a ton of internet threads about them, here are some from the first page of Googling "MLPI"

https://www.reddit.com/r/Minecraft2/comments/1ouzgdr/warning_about_a_minecraft_griefing_group_called/

https://medium.com/@caliasiangirl/how-griefing-groups-are-exploiting-unsecured-minecraft-servers-mlpi-ogmur-5th-column-104c98a372ea

**Reply by teresaleija** [reply]
> hi, I am also interested in this. I have the sever accessible via a public IP and thus anyone with a TLauncher can connect easily. Also via tlaucnher (I was testing) it is also simple to change the uuid in tlauncher.
> 
> Running the server for my son and it seems some guys from "mojang loss prevention" connected somehow (probably first he disabled the whitelist and left the server without whitelist overnight), and now once the usernames are known, it seems so easy to connect to the server.

Fellow parent here. I had to deal with these groups too on my kids Minecraft account, and recently helped another Minecraft player on a separate repo. It seems these are adult men trying to mess around in a child's game, which is, to say the least, quite a sad use of lifetime. 

<img width="1032" height="887" alt="Image" src="https://github.com/user-attachments/assets/668609a3-4ddd-4d2e-86e7-cd9988155a94" />

Mountains of Lava Inc is a known Minecraft griefing group, they go by a couple of other names you can find easily through google, namely MLPI, Mojang Loss Prevention Inc, the 5th Column, as well as Break Blocks Club (aka BreakBlocksClub). Further reading here:

1. https://forums.wynncraft.com/threads/discussion-huge-griefing-wave-by-the-fifth-column-mlpi-dealing-with-ogmur-copenheimer-bots.322980/#:~:text=Hi%20from%20the%20technical%20modding,world%20files%2C%20and%20ruining%20builds.
3. https://gamefaqs.gamespot.com/boards/606524-minecraft/81080328#:~:text=Hey%20everyone%2C%20this%20is%20a,or%20videos%20of%20the%20destruction.
5. https://www.reddit.com/r/Minecraft2/comments/1ouzgdr/warning_about_a_minecraft_griefing_group_called/#:~:text=ogmur%20is%20from%20a%20griefing,players%20can%20join%20your%20world

**Reply by teresaleija** [reply]
I'd also like to add @FreeFallFail is one of the griefers, a part of these groups. The things he mentioned above are entirely true, but I would take any further info with caution. 

<img width="1041" height="911" alt="image" src="https://github.com/user-attachments/assets/50231bf3-6a18-4610-a8ef-03e16af86b25" />

**Comment by itzg** [comment]
I just want to clarify that these are valid concerns, but this is not a challenge that it is unique to this docker image. I'm not a server hosting expert myself, so I would suggest asking around in more general forums like the official Minecraft Discord since people may not see the question here.

**Reply by dgradecak** [reply]
ok, thank you. I will do that and report back here if I find a solution.

I was hoping that there is a way to add an auth server somehow

**Reply by itzg** [reply]
You didn't set online mode to false, right? Otherwise users are already authenticated with Mojang. But again, this is nothing specific to this image.

**Reply by dgradecak** [reply]
no, since some children play with "unofficial" clients/accounts.

I just read about using discordbot and I will try with this plugin https://github.com/DenisIndenbom/DiscordAuth maybe that is a cool thing if it works

**Reply by Homer-Sim** [reply]
Maybe it would be a good idea to setup a wireguard server and the microsoft container is part of this wirguard subnet. Kids could activate the vpn and then connect to your server

**Reply by teresaleija** [reply]
> Maybe it would be a good idea to setup a wireguard server and the microsoft container is part of this wirguard subnet. Kids could activate the vpn and then connect to your server

There's no need. Just get them the full account, and run a command on the Minecraft server doing /whitelist on. I detailed above on these griefing organizations and it seems like this is an easy step to prevent all the hassle of handling the bigots.

**Reply by lucgei231** [reply]
Otherwise use authentic on bukkit/spigot/paper or easy authorization on fabric

**Comment by dgradecak** [comment]
hi, 
coming back to this ... 

I tested the discord auth plugin (with bukkit + discord bot) and made some small fixes in order to correctly work on linux (and this docker image). I find it pretty simple to install and use, and the server is well protected. The kids can administer a private discord channel in order that every user has to click "check"  whenever someone joins the mc server.

While doing this I understood that there a plenty of different mechanism to do authentication for MC with different plugins, but the discord way is pretty straight forward for the "kids gamers".

If needed, although it is pretty simple to get started, I can help on request here

**Reply by Balinus** [reply]
I would be interested to learn how to do that!

**Reply by dgradecak** [reply]
long time have passed ... https://github.com/DenisIndenbom/DiscordAuth that is the plugin I still use. You can ask there but it is quite easy if you have some knowledge about docker. From the top of my head you need to configure a discord bot account, configure the ID/token in the plugin config. I think their docs explain everything

**Reply by Balinus** [reply]
Thanks!

**Comment by hezibenmoshe** [comment]
The loss prevention systems of Mojang have detected piracy. Everyone needs to buy the game.

**Reply by FreeFallFail** [reply]
That is a lie, might as well tell the truth. “Mojang Loss Prevention Inc.” isn’t affiliated with Mojang or any official loss prevention effort; it’s actually the name a group of Minecraft griefers use for themselves online, partnering with the 5th Column. 

Players who’ve dealt with them describe MLPI as an organized griefing group that actively scans for unprotected or cracked/offline-mode servers, joins using spoofed names or automated tools, destroys builds (often by lavacasting large areas), leaves signs or messages linking to their Discord, and even lures devastated server owners with fake “server restoration” help that’s really just trolling or a scam despite their claims of anti-piracy or “renovation,” they’re widely treated as harassers who target vulnerable worlds and mock their victims rather than help them.

There's a ton of internet threads about them, here are some from the first page of Googling "MLPI"

https://www.reddit.com/r/Minecraft2/comments/1ouzgdr/warning_about_a_minecraft_griefing_group_called/

https://medium.com/@caliasiangirl/how-griefing-groups-are-exploiting-unsecured-minecraft-servers-mlpi-ogmur-5th-column-104c98a372ea
